src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersion.java

```diff
@@ -0,0 +1,132 @@
+package com.sap.oss.phosphor.fosstars.model.value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains version tag and release date for an artifact version.
+ */
+public class ArtifactVersion {
+
+  public static final ArtifactVersion EMPTY = new ArtifactVersion("", LocalDate.now());
+
+  private final String version;
+  @JsonDeserialize(using = LocalDateDeserializer.class)
+  @JsonSerialize(using = LocalDateSerializer.class)
+  private final LocalDate releaseDate;
+
+  @JsonIgnore
+  private Optional<SemanticVersion> semanticVersion;
+
+  /**
+   * Initialize the ArtifactVersion based on version tag and release date.
+   *
+   * @param version version tag
+   * @param releaseDate release date
+   */
+  public ArtifactVersion(@JsonProperty("version") String version,
+      @JsonProperty("releaseDate") LocalDate releaseDate) {
+
+    Objects.requireNonNull(version, "Version must be set");
+    Objects.requireNonNull(releaseDate, "Release date must be set");
+
+    this.version = version;
+    this.releaseDate = releaseDate;
+    initSemVer(version);
+  }
+  
+  private void initSemVer(String version) {
+    semanticVersion = SemanticVersion.parse(version);
+  }
+
+  public SemanticVersion getSemanticVersion() {
+    return semanticVersion.orElse(null);
+  }
+
+  @JsonIgnore
+  public boolean isValidSemanticVersion() {
+    return semanticVersion.isPresent();
+  }
+
+  public String getVersion() {
+    return version;
+  }
+
+  public LocalDate getReleaseDate() {
+    return releaseDate;
+  }
+
+  @Override
+  public String toString() {
+    return "{"
+        + "version='" + version + '\''
+        + ", releaseDate=" + releaseDate
+        + '}';
+  }
+
+  /**
+   * LocalDate to Date deserializer used by Jackson Databind for JSON parsing.
+   */
+  private static class LocalDateDeserializer extends StdDeserializer<LocalDate> {
+
+    private static final long serialVersionUID = 1L;
+    private static final Pattern TEST_DATE_PATTERN = Pattern.compile("TEST([+-])(\\d{1,4})d");
```

Comment:
I am not sure I understand why we need this pattern. Is it only for testing purposes? If yes, I'd remove it and add tests with real data. You can have a look at `DateValueTest`.

Reply:
I know that this is a _code smell_. Unfortunately I did not find a solution after I wrote the `OssArtifactSecurityRatingVerificationTest` which uses the yaml approach for test vector definition.
My offer is that I remove the `TEST_DATE_PATTERN` and look for the best way to make the `OssArtifactSecurityRatingVerificationTest` work. Have create #484 for this.

I would propose to keep the test but put a `@Ignore`.
If you do not like this, please let me know then I will delete it.

Reply:
It is okay to open an issue for this.

Reply:
Thanks, have put the test to be ignored for now and created #484